libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Identity/SFIdentityData.m

```diff
@@ -96,55 +97,62 @@ - (NSString *)description
 
 - (NSURL *)idUrl
 {
-    return [NSURL URLWithString:(self.dictRepresentation)[kSFIdentityIdUrlKey]];
+    NSString *value = self.dictRepresentation[kSFIdentityIdUrlKey];
+    return [value isEqual:[NSNull null]] ? nil : [NSURL URLWithString:value];
 }
 
 - (BOOL)assertedUser
 {
-    if ((self.dictRepresentation)[kSFIdentityAssertedUserKey] != nil)
-        return [(self.dictRepresentation)[kSFIdentityAssertedUserKey] boolValue];
-    else
-        return NO;
+    NSNumber* value = self.dictRepresentation[kSFIdentityAssertedUserKey];
+    return [value isEqual:[NSNull null]] ? NO : [value boolValue];
 }
 
 - (NSString *)userId
 {
-    return (self.dictRepresentation)[kSFIdentityUserIdKey];
+    NSString *value = self.dictRepresentation[kSFIdentityUserIdKey];
+    return [value isEqual:[NSNull null]] ? nil : value;
 }
 
 - (NSString *)orgId
 {
-    return (self.dictRepresentation)[kSFIdentityOrgIdKey];
+    NSString *value = self.dictRepresentation[kSFIdentityOrgIdKey];
+    return [value isEqual:[NSNull null]] ? nil : value;
 }
 
 - (NSString *)username
 {
-    return (self.dictRepresentation)[kSFIdentityUsernameKey];
+    NSString *value = self.dictRepresentation[kSFIdentityUsernameKey];
+    return [value isEqual:[NSNull null]] ? nil : value;
 }
 
 - (NSString *)nickname
 {
-    return (self.dictRepresentation)[kSFIdentityNicknameKey];
+    NSString *value = self.dictRepresentation[kSFIdentityNicknameKey];
+    return [value isEqual:[NSNull null]] ? nil : value;
 }
 
 - (NSString *)displayName
 {
-    return (self.dictRepresentation)[kSFIdentityDisplayNameKey];
+    NSString *value = self.dictRepresentation[kSFIdentityDisplayNameKey];
```

Comment:
Revert this function because the value can't be `null`.

Reply:
@bhariharan in the header file you left this as `nullable`, which one is correct?